src/components/layer.js

```diff
@@ -59,13 +78,34 @@ class Layer extends PureComponent {
       zIndex
     } = this.props;
 
-    if (
-      (layerConfig && !isEqual(layerConfig, prevLayerConfig)) ||
-      (params && !isEqual(params, prevParams)) ||
-      (sqlParams && !isEqual(sqlParams, prevSqlParams))
-    ) {
+    // Check that source has changed
+    const prevSourceParsed =
+      prevSource.parse === false
+        ? prevSource
+        : JSON.parse(replace(JSON.stringify(prevSource), prevParams, prevSqlParams));
+
+    const sourceParsed =
+      source.parse === false
+        ? source
+        : JSON.parse(replace(JSON.stringify(source), params, sqlParams));
+
+    // Check that render has changed
+    const prevRenderParsed =
+      prevRender.parse === false
+        ? prevRender
+        : JSON.parse(replace(JSON.stringify(prevRender), prevParams, prevSqlParams));
+
+    const renderParsed =
+      render.parse === false
+        ? render
+        : JSON.parse(replace(JSON.stringify(render), params, sqlParams));
+
+    if (sourceParsed && !isEqual(sourceParsed, prevSourceParsed)) {
```

Comment:
Ideally we should save the source after parsing, this way we can do a simple : `Object.is(source, prevSource)` or a shallow compare:
```js
Object.keys(source).forEach(key => Object.is(source[key], prevSource[key])
```

Reply:
As we talk yesterday, we can't do this at this point. We should save it in the model parsed, that's true. We will think of a way of doing it